icu4j/tools/misc/src/main/java/com/ibm/icu/dev/tool/ime/indic/IndicInputMethodImpl.java

```diff
@@ -373,7 +373,7 @@ public int getRunStart() {
 
         public int getRunStart(AttributedCharacterIterator.Attribute attribute) {
             return (index >= committed &&
-                attribute == TextAttribute.INPUT_METHOD_UNDERLINE) ? committed : 0;
+                attribute.equals(TextAttribute.INPUT_METHOD_UNDERLINE)) ? committed : 0;
```

Comment:
- This is in .../tools/misc/.../dev/tool/ime/... so probably no one uses it.
- .equals() should be fine
- However, it is a little weird having to use .equals() for a `static final` constant -- an immutable singleton.
  https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/java/awt/font/TextAttribute.html#INPUT_METHOD_UNDERLINE

Reply:
I can "sprinkle" `@SuppressWarnings("ReferenceEquality")`

Even if it is safe to use `==`, that is not good enough for the linter.
It is not smart enough to "know" that.
All I can do is use `.equals`, or disable the warning.
You know how linters work :-)
If we want to enable this check in CI (and fail if new instances are introduced) then this must be "solved" (even if we know it is not a problem).

This being some (very much outdated) GUI tool (which most of the time waits for the user), I thought the performance is not a problem.

Reply:
TLDR: I don't feel strongly about `SuppressWarnings` or `.equals`. Coin toss.